### PR TITLE
ENG-4738 add disableContentMenu property

### PR DESCRIPTION
**Pull request by gp-ent**
Add the `disableContentMenu` boolean property to `userPreferences` api.

Liquibase changelog to create the new column is provided but I've some dubts about the positioning of the changelog file itself. I put it in the `engine` module because that's were the schema is created. Should I have put in the `webapp` module? Note that the changelog has no data, just the column creation with `null` value.

### CODE REVIEW

File: engine/src/main/java/org/entando/entando/web/userpreferences/model/UserPreferencesDto.java [hunk 1]
```diff
@@ -149,6 +151,14 @@ public void setDefaultWidgetJoinGroups(List<String> defaultWidgetJoinGroups) {
         this.defaultWidgetJoinGroups = defaultWidgetJoinGroups;
     }
 
+    public Boolean getDisableContentMenu() {
```
Comment (entgigi) [inline engine/src/main/java/org/entando/entando/web/userpreferences/model/UserPreferencesDto.java]: You should use lombok to reduce code duplication

File: engine/src/main/java/org/entando/entando/web/userpreferences/model/UserPreferencesRequest.java [hunk 2]
```diff
@@ -99,6 +100,14 @@ public void setDefaultWidgetJoinGroups(List<String> defaultWidgetJoinGroups) {
         this.defaultWidgetJoinGroups = defaultWidgetJoinGroups;
     }
 
+    public Boolean getDisableContentMenu() {
```
Comment (entgigi) [inline engine/src/main/java/org/entando/entando/web/userpreferences/model/UserPreferencesRequest.java]: you should use lombok to reduce code duplication

File: engine/src/main/resources/liquibase/changeSetPort.xml [hunk 3]
```diff
@@ -23,4 +23,7 @@
 
     <include file="port/20230308000000_legacy_api_refactoring.xml" relativeToChangelogFile="true" />
 
-</databaseChangeLog>
+    <include file="port/20230407000000_disable_content_menu.xml" relativeToChangelogFile="true" />
```
Comment (entgigi) [inline engine/src/main/resources/liquibase/changeSetPort.xml:26]: the choice to place this stuff in the engine module seems good to me, please @eugeniosant can you check too?

**Review by entgigi — CHANGES_REQUESTED** [review]

File: engine/src/main/java/org/entando/entando/web/userpreferences/model/UserPreferencesDto.java (hunk 1 above)
Comment (gp-ent) [reply engine/src/main/java/org/entando/entando/web/userpreferences/model/UserPreferencesDto.java]: @entgigi implementing lombok didn't reduce the code duplication. The problem is that `UserPreferencesRequest` and `UserPreferencesDto` are almost identical, to reduce the duplication they should be merged into a single class but, as discussed with @eugeniosant, that kind of change is out of scope right now

File: engine/src/main/java/org/entando/entando/web/userpreferences/model/UserPreferencesRequest.java (hunk 2 above)
Comment (gp-ent) [reply engine/src/main/java/org/entando/entando/web/userpreferences/model/UserPreferencesRequest.java]: see previous comment

**Review by entgigi — APPROVED** [review]
LGTM

**Comment by gp-ent** [comment]
As discussed with @eugeniosant we decided to ignore the quality gate failing on code duplication due to the small impact of the new code.

**Pull request merged**